### DISCUSSION TITLE
Please kindly help how to define Crystal frequency on ESP8266?

### DISCUSSION THREAD

**Discussion by hamwong** [arendst/Tasmota#12928] [Q&A] [answered]
I got bunch of Huafan_SS Devices which runs on ESP8266 with crystal frequency 40Mhz, I used to compile by Arduino, but I am fail to compile 8.5.1, please kindly advise how can I define crystal Freq on gitpod or Platformio with vsc?
I have google around for 2 days, appreciated for any help

**Comment by hamwong** [comment, marked as answer]
I have found the way, thanks for vic, I finally realize what you said on discord,
rename **platformio_override_sample.ini** to **platformio_override.ini**

modify line 48 (few lines below [common])
build_flags                 = ${core.build_flags}    >>>>> to >>>>>   build_flags                 = ${core.build_flags}  -DF_CRYSTAL=40000000

**Reply by jwrw** [reply]
You're a star! - I'd not updated my Huafans since writing the original article and was finding the board_build.f_cpu  setting (that ought to do the right thing) wasn't working.  This fix seems to have done the trick and I can now get my sockets working again!
